lexos/models/stats_model.py

```diff
@@ -0,0 +1,179 @@
+    hapax: int  # The hapax of all word counts of a file.
+    median: float  # The median of all word counts of a file.
+    average: float  # The average of all word counts of a file.
+    num_word: int  # The number of words of a file.
+    file_name: str  # The name of a file.
+    word_count: int  # The count of all words within a file.
+    std_deviation: float  # The standard deviation of word counts.
+    total_word_count: int  # The total of all word counts of a file.
+
+
+class StatsModel(BaseModel):
+    def __init__(self, test_options: Optional[StatsTestOptions] = None):
+        """This is the class to generate statistics of the input file.
+
+        :param test_options: the input used in testing to override the
+                             dynamically loaded option
+        """
+        super().__init__()
+        if test_options is not None:
+            self._test_dtm = test_options.doc_term_matrix
+            self._test_id_temp_label_map = test_options.id_temp_label_map
+        else:
+            self._test_dtm = None
+            self._test_id_temp_label_map = None
+
+    @property
+    def _doc_term_matrix(self) -> pd.DataFrame:
+        """:return: the document term matrix"""
+        return self._test_dtm if self._test_dtm is not None \
+            else MatrixModel().get_matrix()
+
+    @property
+    def _id_temp_label_map(self) -> IdTempLabelMap:
+        """:return: a map takes an id to temp labels"""
+        return self._test_id_temp_label_map \
+            if self._test_id_temp_label_map is not None \
+            else MatrixModel().get_temp_label_id_map()
+
+    def _get_corpus_info(self) -> CorpusInfo:
+        """Converts word lists completely to statistic."""
+        assert np.sum(self._doc_term_matrix.values) > 0, EMPTY_LIST_MESSAGE
+        # initialize
+        file_anomaly_iqr = {}
+        file_anomaly_std_err = {}
+        num_file = np.size(self._doc_term_matrix.index.values)
+        file_sizes = np.sum(self._doc_term_matrix.values, axis=1)
+        labels = [self._id_temp_label_map[file_id]
+                  for file_id in self._doc_term_matrix.index.values]
+
+        # 1 standard error analysis
+        average_file_size = round(np.average(file_sizes), 3)
+        # Calculate the standard deviation
+        std_dev_file_size = np.std(file_sizes).item()
+        # Calculate the anomaly
+        for count, label in enumerate(labels):
+            if file_sizes[count] > average_file_size + 2 * std_dev_file_size:
+                file_anomaly_std_err.update({label: 'large'})
+            elif file_sizes[count] < average_file_size - 2 * std_dev_file_size:
+                file_anomaly_std_err.update({label: 'small'})
+
+        # 2 iqr analysis
+        median = np.median(file_sizes).item()
+        q1 = np.percentile(file_sizes, 25, interpolation="midpoint")
+        q3 = np.percentile(file_sizes, 75, interpolation="midpoint")
+        iqr = q3 - q1
```

Comment:
For example, what is iqr? I think q1 and q3 are quartiles, but you don't say this for certain

Reply:
These were explained when I implemented the namedTuple... But I guess it was a bad idea...